src/test/java/mate/academy/dao/impl/UserDaoImplTest.java

```diff
@@ -0,0 +1,72 @@
+package mate.academy.dao.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.RoleDao;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.Role.RoleName;
+import mate.academy.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserDaoImplTest extends AbstractTest {
```

Comment:
What if I have to save existing user or save user with null email. I think it should be covered by tests